app/_features/event/components/event-form.tsx

```diff
@@ -0,0 +1,493 @@
+import { useNavigate } from '@/_shared/hooks/use-navigate';
+import ClockFillIcon from '@/_shared/components/icons/clock-fill-icon';
+import PhotoUploadIcon from '@/_shared/components/icons/photo-upload-icon';
+import { PhotoDeleteIcon } from '@/_shared/components/icons/photo-delete-icon';
+import { ActionType, ImageCropperModal, ImageState } from './image-cropper';
+import { selectEvent } from '@/(server)/_features/event/schema';
+import { compressImage } from '@/_shared/utils/compress-image';
+import SignInModal from '@/_shared/components/auth/sign-in-modal';
+
+const reducer = (state: ImageState, action: ActionType): ImageState => {
+  switch (action.type) {
+    case 'ConfirmCrop':
+      return {
+        imagePreview: action.payload.preview,
+        imageBlob: action.payload.blob,
+      };
+    case 'PickFile':
+      return { imagePreview: null, imageBlob: action.payload };
+    case 'Reset':
+      return { imagePreview: null, imageBlob: null };
+    default:
+      return state;
+  }
+};
+
+const APP_ORIGIN = process.env.NEXT_PUBLIC_APP_ORIGIN;
+const LOCAL_STORAGE_PATH = process.env.LOCAL_STORAGE_PATH;
+
+export default function EventForm({
+  data: existingEvent,
+}: {
+  data?: typeof selectEvent;
+}) {
+  // Constant for event
+  const setStartTimeEvent = 'open:event-time-picker-start-modal';
+  const setEndTimeEvent = 'open:event-time-picker-end-modal';
+  const comingSoonEvent = 'open:feature-coming-soon-modal';
+  const incompleteFieldEvent = 'open:missing-field-modal';
+
+  // States
+  const { user } = useAuthContext();
+  const [imageData, updateImageData] = useReducer(reducer, {
+    imageBlob: null,
+    imagePreview: null,
+  });
+  const today = new Date(new Date().setDate(new Date().getDate() + 1));
+  const currentHour = today.getHours();
+  const [date, setDate] = useState(today);
+  const [startTime, setStartTime] = useState({
+    hour: `${currentHour}`,
+    minute: `${today.getMinutes().toString().padStart(2, '0')}`,
+  });
+  const [endTime, setEndTime] = useState({
+    hour: `${currentHour + 1}`,
+    minute: `${today.getMinutes().toString().padStart(2, '0')}`,
+  });
+  const [isTitleValid, setTitleValid] = useState(false);
+  const [isDescriptionValid, setDescValid] = useState(false);
+  const [eventName, setEventName] = useState('');
+  const [eventDescription, setEventDescription] = useState('');
+
+  const { navigateTo } = useNavigate();
+
+  const handleRemoveImage = useCallback(() => {
+    updateImageData({ type: 'Reset' });
+  }, []);
+
+  const handleFileSelect = useCallback(
+    (event: ChangeEvent<HTMLInputElement>) => {
+      const file = event.target.files && event.target.files[0];
+
+      if (file) {
+        updateImageData({ type: 'PickFile', payload: file });
+
+        document.dispatchEvent(new CustomEvent('open:image-cropper'));
+      }
+    },
+    []
+  );
+
+  useEffect(() => {
+    if (!user) document.dispatchEvent(new CustomEvent('open:sign-in-modal'));
+  }, [user]);
```

Comment:
You shouldn't check if the user has logged in or not in this component. This component should be only rendered/displayed when the user actually exists. This logic should be outside this component. For example, when the user is not logged in, the user should not be able to access the create event or update event pages.

Reply:
I've limited the access to the page by showing a http error instead https://github.com/inlivedev/inlive-room/pull/130/commits/7c9946511cbc67deaeae121f6a1f4d4d4935bc88